src/storage-controller/src/history.rs

```diff
@@ -80,6 +81,7 @@ impl<T: std::fmt::Debug> CommandHistory<T> {
 
         let mut create_timely_command = None;
         let mut initialization_complete = false;
+        let mut read_only = true;
```

Comment:
Nit: I'd call this `allow_writes` for a slight readability boost.

Reply:
Ah sure, done. I was matching the compute controller history reduction implementation, but I do agree calling this `allow_writes` is cleaner. I figure you'll make the compute controller history reduction match this phrasing at some point. :)